### PR TITLE
Change all md files extension to mdx

### PR DESCRIPTION
**Pull request by arshadkazmi42**
Bugzilla Bug: [1530790](https://bugzilla.mozilla.org/show_bug.cgi?id=1530790)

Releated to #1006

### CODE REVIEW

**Comment by arshadkazmi42** [comment]
@helfi92 Some of the tests are failing. Did I made a wrong change somewhere?
Does this PR needs only file extension change from md to mdx?
As I have updated .md to .mdx where ever it was getting used in tests, test data.

File: README.md [hunk 1]
```diff
@@ -75,12 +75,11 @@ It is not possible to run a full Taskcluster deployment directly from this repos
     * [Web-Server Service](services/web-server#readme)
     * [Worker Manager Service](services/worker-manager#readme)
         * [About the Keys in this  Directory](services/worker-manager/src/providers/aws-keys#readme)
-* [Taskcluster UI](ui#readme)
```
[the remark this answers is not in the document]
Comment (helfi92) [reply README.md]: That's probably because of `ui/README` being changed to `.mdx`.

**Comment by helfi92** [comment]
@arshadkazmi42 I wonder if something went wrong in the rebase. I am seeing a lot of other commits.

**Comment by arshadkazmi42** [comment]
@helfi92 I did 
- `git fetch upstream`
- `git rebase upstream/master`
- `git checkout my-branch`
- `git rebase master`

**Comment by helfi92** [comment]
Hm, I think that should of worked. Try doing `git pull upstream master --rebase` and then force pushing your changes here. If that doesn't do it, it shouldn't be a problem if you open a new PR.

**Comment by arshadkazmi42** [comment]
> Hm, I think that should of worked. Try doing `git pull upstream master --rebase` and then force pushing your changes here. If that doesn't do it, it shouldn't be a problem if you open a new PR.

Did rebasing again. Looks like this has removed those commits.

File: ui/docs/people.mdx [hunk 2]
```diff
@@ -7,7 +7,7 @@ import People from 'taskcluster-ui/views/Documentation/People';
 # People of Taskcluster
 
 Thanks goes to these wonderful people for their hard work on Taskcluster.
-Your name [could be here, too](https://github.com/taskcluster/taskcluster/blob/master/CONTRIBUTING.md)!
+Your name [could be here, too](https://github.com/taskcluster/taskcluster/blob/master/CONTRIBUTING.mdx)!
```
Comment (djmitche) [inline ui/docs/people.mdx]: This file did not get renamed (nor should)

File: ui/docs/reference/core/web-server/README.mdx [hunk 3]
```diff
@@ -23,8 +23,8 @@ Again, this approach is intended only for use by the Taskcluster UI.
 
 The following login strategies are supported:
 
-* [GitHub](github-login-strategy.md)
-* [Mozilla-Auth0](mozilla-auth0-login-strategy.md)
+* [GitHub](github-login-strategy.mdx)
+* [Mozilla-Auth0](mozilla-auth0-login-strategy.mdx)
```
Comment (djmitche) [inline ui/docs/reference/core/web-server/README.mdx]: Interesting -- in fact these should not have extensions at all (just the filename)

Comment (arshadkazmi42) [reply ui/docs/reference/core/web-server/README.mdx]: I think these are pointing to these files
https://github.com/taskcluster/taskcluster/blob/master/ui/docs/reference/core/web-server/github-login-strategy.md
https://github.com/taskcluster/taskcluster/blob/master/ui/docs/reference/core/web-server/mozilla-auth0-login-strategy.md

These files are renamed to `mdx` in this PR

File: ui/docs/people.mdx (hunk 2 above)
Comment (arshadkazmi42) [reply ui/docs/people.mdx]: Done

Comment (helfi92) [reply ui/docs/people.mdx]: Hm, did you forget to push the commit by any chance? :)

Comment (arshadkazmi42) [reply ui/docs/people.mdx]: > Hm, did you forget to push the commit by any chance? :)

I haven't pushed the changes. Waiting for reply for the second comment. Will be pushing both the fixes together.

File: ui/docs/reference/core/web-server/README.mdx (hunk 3 above)
Comment (helfi92) [reply ui/docs/reference/core/web-server/README.mdx]: What happens if you remove the `.mdx` extension. Does the page still render?

Comment (arshadkazmi42) [reply ui/docs/reference/core/web-server/README.mdx]: It shows 404 page on clicking on any of the option if we change names to `md`

Comment (helfi92) [reply ui/docs/reference/core/web-server/README.mdx]: I don't seem to be able to view the links on the preview site. Can you replicate?

https://deploy-preview-1331--taskcluster-web.netlify.com/docs/reference/core/web-server

Comment (djmitche) [reply ui/docs/reference/core/web-server/README.mdx]: The existing `.md` links don't work, either -- the links are already broken before this PR.  All of the links between pages on the docs site omit the extension, and whoever wrote this (probably me) forgot that and didn't check the links before making a PR (yeah, sounds just like me).

Comment (arshadkazmi42) [reply ui/docs/reference/core/web-server/README.mdx]: https://deploy-preview-1331--taskcluster-web.netlify.com/docs/reference/core/web-server/github-login-strategy

https://deploy-preview-1331--taskcluster-web.netlify.com/docs/reference/core/web-server/mozilla-auth0-login-strategy

These are the links on preview site and it seems working

I have updated the PR with removed mdx extension

Comment (arshadkazmi42) [reply ui/docs/reference/core/web-server/README.mdx]: https://deploy-preview-1331--taskcluster-web.netlify.com/docs/reference/core/web-server/github-login-strategy

https://deploy-preview-1331--taskcluster-web.netlify.com/docs/reference/core/web-server/mozilla-auth0-login-strategy

These are the links on preview site and it seems working

I have updated the PR with removed mdx extension

**Pull request merged**